app/src/main/java/com/keylesspalace/tusky/util/AccountPreferenceDataStore.kt

```diff
@@ -0,0 +1,23 @@
+package com.keylesspalace.tusky.util
+
+import androidx.preference.PreferenceDataStore
+import com.keylesspalace.tusky.db.AccountEntity
+import com.keylesspalace.tusky.settings.PrefKeys
+
+class AccountPreferenceDataStore(private val account: AccountEntity): PreferenceDataStore() {
```

Comment:
Once such a solution is introduced it might probably be sensible to remove the respective settings from the shared preferences.

There they will be confusing now and have no meaning in the future.

Reply:
Also see #3186 for this.

Reply:
Updated that migration code to the new settings migration mechanism.